rust-runtime/aws-smithy-types/src/body.rs

```diff
@@ -139,6 +142,24 @@ impl SdkBody {
         poll_fn(|cx| me.as_mut().poll_next(cx)).await
     }
 
+    #[cfg(feature = "http-body-1-x")]
+    pub(crate) fn poll_next_frame(
+        self: Pin<&mut Self>,
+        cx: &mut Context<'_>,
+    ) -> Poll<Option<Result<http_body_1_0::Frame<Bytes>, Error>>> {
+        // There has got to be a way to simplify this matching matchy match
+        match self.poll_next(cx) {
+            Poll::Pending => return Poll::Pending,
+            Poll::Ready(maybe_ready) => match maybe_ready {
+                None => return Poll::Ready(None),
+                Some(result) => match result {
+                    Err(err) => return Poll::Ready(Some(Err(err))),
+                    Ok(bytes) => return Poll::Ready(Some(Ok(http_body_1_0::Frame::data(bytes)))),
+                },
+            },
+        }
+    }
```

Comment:
this doesn't handle trailers correctly. I'd start by adding a test case involving trailers (that should fail), then go from there.